Modules/Devices/src/Devices.Infrastructure/Persistence/Repository/OAuthClientsRepository.cs

```diff
@@ -17,7 +20,30 @@ public OAuthClientsRepository(OpenIddictApplicationManager<OpenIddictEntityFrame
     public async Task<IEnumerable<OAuthClient>> FindAll(CancellationToken cancellationToken)
     {
         var clients = await _applicationManager.ListAsync(applications => applications.Select(c => new OAuthClient(c.ClientId, c.DisplayName)), cancellationToken).ToListAsync(cancellationToken);
-
         return clients;
     }
+
+    public async Task<OAuthClient> Find(string clientId, CancellationToken cancellationToken)
+    {
+        var client = await _applicationManager.FindByClientIdAsync(clientId, cancellationToken);
+        return client != null ? new OAuthClient(client.ClientId, client.DisplayName) : null;
+    }
+
+    public async Task Add(string clientId, string displayName, string clientSecret, CancellationToken cancellationToken)
+    {
+        var managerResult = await _applicationManager.CreateAsync(new OpenIddictApplicationDescriptor
+        {
+            ClientId = clientId,
+            ClientSecret = clientSecret,
+            DisplayName = displayName,
+            Permissions =
+            {
+                Permissions.Endpoints.Token,
+                Permissions.GrantTypes.Password
+            }
+        }, cancellationToken);
+
+        if (managerResult == null)
+            throw new Exception($"Failed to create the client: '{displayName}'");
```

Comment:
Add the client id to the exception, so that we can comprehend the error from the logs.